packages/yew-macro/src/derive_props/field.rs

```diff
@@ -320,7 +318,29 @@ fn is_path_segments_an_option(path_segments: impl Iterator<Item = String>) -> bo
 ///
 /// Users can define their own [`Option`] type and this will return true - this is unavoidable.
 fn is_path_an_option(path: &Path) -> bool {
-    is_path_segments_an_option(path.segments.iter().take(3).map(|ps| ps.ident.to_string()))
+    is_path_segments_an_option(path.segments.iter().map(|ps| &ps.ident))
+}
+
+fn is_path_segments_a_string<'a, T>(mut iter: impl Iterator<Item = &'a T>) -> bool
+where
+    T: 'a + ?Sized + PartialEq<str>,
+{
+    iter.next().map_or(false, |first| {
+        first == "String"
+            || (first == "std" || first == "alloc")
+                && iter.next().map_or(false, |x| x == "string")
+                && iter.next().map_or(false, |x| x == "String")
+    })
```

Comment:
I think adding an early return might help to make the conditions chain easier to read.

```suggestion
    iter.next().map_or(false, |first| {
        if first == "String" {
            return true;
        }

        (first == "std" || first == "alloc")
            && iter.next().map_or(false, |x| x == "string")
            && iter.next().map_or(false, |x| x == "String")
    })
```

Also, do you think could be a good idea to add some names like `second` and `third`?


```suggestion
    iter.next().map_or(false, |first| {
        first == "String"
            || (first == "std" || first == "alloc")
                && iter.next().map_or(false, |second| second == "string")
                && iter.next().map_or(false, |third| third == "String")
    })
```

What do you think?

Reply:
I believe the `||` operator is perfectly readable and makes the function concise, but I do agree that giving descriptive names to closures' arguments is a worthwhile measure, I'll add them right away